### PR TITLE
docs: add an explanation about prefetch on hover to the link.mdx

### PR DESCRIPTION
**Pull request by suu3**
Hello, I noticed the following code in `link.tsx`.
```
//client/link.tsx
...
onMouseEnter(e) {
    ...
    if (
        (!prefetchEnabled || process.env.NODE_ENV === 'development') &&
        isAppRouter
    ) {
        return
    }

    prefetch(
          ...
    )
},
```
It seems that when prefetch={false} is set, prefetch still occurs on hover for page routers.
However, there is no explanation in the <Link> documentation, which caused misleading.
I added the explanation inside PageOnly component, could you please review it?

### CODE REVIEW

File: docs/02-app/02-api-reference/01-components/link.mdx [hunk 1]
```diff
@@ -220,7 +220,9 @@ export default function Page() {
 Prefetching happens when a `<Link />` component enters the user's viewport (initially or through scroll). Next.js prefetches and loads the linked route (denoted by the `href`) and data in the background to improve the performance of client-side navigations. Prefetching is only enabled in production.
 
 - **`true` (default)**: The full route and its data will be prefetched.
-- `false`: Prefetching will be disabled.
+- `false`: Prefetching will be disabled. 
+
+> **Good to know**: When `prefetch` is set to `false`, prefetching will still occur on hover.
```
Comment (karimxload) [inline docs/02-app/02-api-reference/01-components/link.mdx:225]: Isn't prefetching will still occur on hover in development mode ?

Comment (samcx) [reply docs/02-app/02-api-reference/01-components/link.mdx:225]: @karimxload It does not on either Pages or App Router in development mode.

**Review by samcx — APPROVED** [review]
Thank you for submitting a <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 57fe86341889e1f86f5cee460c49993d224d21df

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by samcx** [comment]
@suu3 No need to merge the latest canary!

**Pull request merged**